examples/systemcalls/systemcalls.c

```diff
@@ -92,6 +156,56 @@ bool do_exec_redirect(const char *outputfile, int count, ...)
  *   The rest of the behaviour is same as do_exec()
  *
 */
+    int fd = open(outputfile, (O_RDWR | O_CREAT | O_TRUNC),
+           (S_IWUSR | S_IRUSR | S_IWGRP | S_IRGRP | S_IROTH ));
+
+    if (fd == -1)
+    {
+        return false;
+    }
+
+    pid = fork();
```

Comment:
@Buff-Chandan think about how you could avoid duplicating logic with the function above.